test/blackbox-tests/test-cases/pkg/additional-constraints-ocaml-system.t

```diff
@@ -0,0 +1,57 @@
+Test that we support the use case of using additional constraints to force the
+system compiler to be used instead of installing the compiler.
+
+  $ . ./helpers.sh
+
+Create packages resembling the ocaml compiler packages
+  $ mkrepo
+
+  $ mkpkg ocaml-base-compiler <<EOF
+  > conflicts: [ "ocaml-system" ]
+  > EOF
+
+  $ mkpkg ocaml-variants <<EOF
+  > conflicts: [ "ocaml-system" ]
+  > EOF
+
+  $ mkpkg ocaml-system
```

Comment:
shouldn't the system compiler have a conflict with ocaml-base-compiler and ocaml-variants?

Reply:
ah yeh you're right. All the compiler options should conflict with one another.

Reply:
fixed